Reduce-made-easy.md

```diff
@@ -0,0 +1,370 @@
+These two arguments are called in the official documentation as `prev` and `next`. Personally, I don't think those names do justice to their true nature.
+
+So, throughout this text, we would be referring to them as `acc`, to represent accumulated value; and `item`, to denote the current item being accessed.
+
+With these so far, here's what a `reduce` should look like:
+
+```js
+arr.reduce(function(acc, item){
+ /* here you have to complete the function */
+}, initialValue);
+```
+
+Now, let's find out what would the value of these `acc` and `item` are. We have mentioned earlier that the `reduce` is a replacement for itertaive constructs.
+
+It stands to reason that `reduce` would take your custom callback function; and iterate over the Array on which `reduce` has been invoked.
+
+Instead of describing these, let's ask the JS execution engine what these are!
+
+```js
+var arr = [10, 20, 30, 60];
+arr.reduce(function(acc, item){
+   console.log(acc, item);
+}, 0);
+```
+Executing the above in browser or Node console would give you this as output:
+
+```js
+0 10
+undefined 20
+undefined 30
+undefined 60
+```
+
+Notice the number of outputs are same as number of elements in the Array `[10, 20, 30, 60]`. In fact, it prints out the elements of the Array!
+
+So, we can deduce that `reduce()` takes your custom callback and executes it on each element of the Array. While doing so, it makes the current item available to the custom callback as `item` argument.
+
+But what about `acc`? We see that other than the first line, when `item = 10`, it is `undefined`. In the first line, which corresponds to the first iteration, its value is same as `initialvalue`.
+
+In short, our `acc` accumulator, is not accumulating!
+
+But then, how do we make it accumulate? Let's try executing this:
+
+```js
+var arr = [10, 20, 30, 60];
+arr.reduce(function(acc, item){
+   console.log(acc, item);
+   return acc;
+}, 0);
+```
+
+This time the output changes to:
+
+```js
+0 10
+0 20
+0 30
+0 60
+```
+
+As you can see, the value of `acc` would remain constant throughout. And that is expected - we are not altering the value of `acc` anywhere in the custom callback. We return whatever `reduce` makes available at a given iteration.
+
+But we did realize something - the value of `acc` for current iteration, would be the `return` value from custom callback from previous iteration. And ultimately, when the iteration is over; the final value of `acc` would be returned by `reduce` call.
+
+This leaves only one important part in our understanding - the value of execution context, or `this`!
+
+So, we again approach our friendly neighbor, the JS console and execute this:
+
+```js
+var arr = [10, 20, 30, 60];
+arr.reduce(function(acc, item){
+   console.log(acc, item, this);
+   return acc;
+}, 0);
+```
+
+If you are in strict mode, it would return `undefined` as value of `this`. Otherwise, in-browser, it would point to `window` object as `this`.
```

Comment:
Can we point to the MDN link in the text "strict mode"? 
It would be a good clarification for someone who isn't aware of it.

Apart from that this is an excellent write up. Just [squash](https://github.com/FreeCodeCamp/wiki/blob/master/git-rebase.md#squashing-multiple-commits-into-one) the commits and we are good to go!

Thanks for the article @alayek !